apps/vast/message_mapping.py

```diff
@@ -78,14 +78,9 @@ def to_vast_query(intel: Intel):
     if vast_type == "ip":
         return str(indicator)
     if vast_type == "url":
-        # TODO: use field annotations, once implemented (ch17531)
-        return f'"{indicator}" in url'
+        return f'"{indicator}" in net.uri'
     if vast_type == "domain":
-        # Currently, uses VAST's suffix-based field matching. Targets every
-        # schema with fieldnames that end in `domain`, `host`, or `hostname`
-        # (Zeek and Suricata)
-        # TODO: use field annotations, once implemented (ch17531)
-        return f'"{indicator}" in domain || "{indicator}" in host || "{indicator}" in hostname'
+        return f'"{indicator}" in net.domain || "{indicator}" in net.hostname'
```

Comment:
Just a general question here: how does that work with escaping? Can we have a unit test that checks that?

Reply:
What is the issue with escaping, can you give an example?

When you look at the two files that this PR touches, you'll see there already is a unit test.

Reply:
Consider the indicator `e"il.com`. If you do not escape it properly, then the query expression is invalid. For such a corner case, it would make sense to have a unit test to ensure that this gets converted properly to `e\"il.com` before the interpolation takes place.

Reply:
As discussed in the office, I created new story for this.

Reply:
Review re-requested.